### PR TITLE
Report goal  adva

### PR DESCRIPTION
**Pull request by AtomicManic**
After efactor changes, both on the report goal and the main flow

### CODE REVIEW

File: controller/external.controller.js [hunk 1]
```diff
@@ -85,4 +90,28 @@ const checkIfExperimentIsActive = async (experimentId) => {
   return true;
 };
 
-module.exports = { runTest };
+const reportGoal = async (req, res) => {
```
Comment (sufka861) [inline controller/external.controller.js:93]: looks good

File: middleware/errorHandler.mw.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const logger = require("./../middleware/loggers/error.logger");
 
 exports.errorHandler = (error, req, res, next) => {
+  console.log(error);
```
Comment (sufka861) [inline middleware/errorHandler.mw.js:4]: should be removed, i assume it was there for debug

File: models/Experiment.model.js [hunk 3]
```diff
@@ -1,122 +1,124 @@
...
+      type: String,
+      enum: ["f-f", "a-b"],
+      required: true,
+    },
+    testAttributes: {
+      location: {
+        type: [String],
+        validate: {
+          validator: countryValidator,
+          message: () => `Invalid country code`,
```
Comment (sufka861) [inline models/Experiment.model.js:18]: did you make any changes to the Experiment.model??
for some reason the code dosent exactly line up next to each other so it is difficult to compare

**Review by sufka861 — APPROVED** [review]
looks good, made a few comments.
didn't exactly see if there were changes to experiment.model other than spacing and such?

File: middleware/errorHandler.mw.js (hunk 2 above)
Comment (AtomicManic) [reply middleware/errorHandler.mw.js:4]: you are right, will do

File: models/Experiment.model.js (hunk 3 above)
Comment (AtomicManic) [reply models/Experiment.model.js:18]: I wrote down the changes everyone should make in lines 1 and 94. It was after I've talk to you and peer yesterday, he made some minor mistakes with the schema. Also Shir made her refactor changes, i wrote about it yesterday on slack

**Pull request merged**